src/js/angular/getImgService.test.js

```diff
@@ -0,0 +1,134 @@
+                expect(ImgService.loadPics).to.exist();
+            });
+            it('should use $http to retrieve data', function(){
+                var spy = sinon.spy($http, 'get');
+                var method = "flickr.photos.search";
+                var api_key = '74e47a159e15cbcb6139ba9c9df64c13';
+                var user_id = "138698049@N03";
+                var mockUrl = 'https://api.flickr.com/services/rest?method=' + method + '&api_key='+ api_key + '&user_id=' + user_id + '&format=json&nojsoncallback=1';
+
+                ImgService.loadPics();
+                expect(spy).to.have.been.calledWith(mockUrl);
+            });
+
+            it('should return a promise', function(){
+               expect(ImgService.loadPics().then).to.exist();
+            });
+        });
+    });
+    describe('createImagesArray', function(){
+        var sampleJSON = {contentType: "text/json",
+            status: "success",
+
+            photos: {
+                photo: [
+                    {
+                        farm: 6,
+                        id: "23780914905",
+                        isfamily: 0,
+                        isfriend: 0,
+                        ispublic: 1,
+                        owner: "138698049@N03",
+                        secret: "0eb2a77a55",
+                        server: "5764",
+                        title: ""
+                    },
+                    {
+                        farm: 1,
+                        id: "23698470321",
+                        isfamily: 0,
+                        isfriend: 0,
+                        ispublic: 1,
+                        owner: "138698049@N03",
+                        secret: "1d4dc255d0",
+                        server: "693",
+                        title: ""
+                    }
+                ]
+            }
+
+
+    };
+
+        describe('should check the data taken by createdDomElm', function(){
+            var getImages = function(){
+                return ImgService.createImagesArray(sampleJSON);
+            };
+
+            it('should exist', function(){
+                expect(ImgService.createImagesArray).to.exist();
+            });
+            it('should return an array', function(){
+                expect(Array.isArray(getImages())).to.be.true();
+            });
+            it('should not be undefined', function(){
+                expect(getImages()).should.not.be.undefined();
+            });
+            it('should not be null', function(){
+                expect(typeof(getImages())).should.not.be.null();
+            });
+            it('should get the length of the object', function(){
+                expect(getImages().length).to.equal(2);
+            });
+        });
+
+        describe('should create an array with urls inside of it', function(){
+            it('should return something', function(){
+                expect(typeof(ImgService.createImagesArray(sampleJSON))).to.not.be.null();
+            });
```

Comment:
Nice check here. Now we know that it SHOULD NOT be `null`, could you write another test to ensure the returned value is the proper type? (**hint:** angular.isArray() returns true/false value)